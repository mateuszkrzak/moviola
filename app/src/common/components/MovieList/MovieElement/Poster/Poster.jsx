import React from 'react';
import PropTypes from 'prop-types';

import './Poster.scss';

export default class Poster extends React.Component {
  static defaultProps = {
    title: 'No title provided',
    src: 'http://via.placeholder.com/500x750 ',
  };

  static propTypes = {
    title: PropTypes.string,
    src: PropTypes.string,
  };

  render() {
    return <img styleName="poster" src={this.props.src} alt={`Poster of ${this.props.title}`} />;
  }
}
