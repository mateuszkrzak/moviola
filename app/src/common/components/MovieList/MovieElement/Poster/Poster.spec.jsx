import { shallow } from 'enzyme';
import React from 'react';
import Poster from './Poster';

test('Poster component renders placeholder if no passed props', () => {
  const wrapper = shallow(<Poster />);
  expect(wrapper).toMatchSnapshot();
});

test('Poster component renders release year', () => {
  const mockPosterProps = { src: 'http://url.for.test.title.cover', title: 'Test title' };
  const wrapper = shallow(<Poster {...mockPosterProps} />);
  expect(wrapper).toMatchSnapshot();
});
