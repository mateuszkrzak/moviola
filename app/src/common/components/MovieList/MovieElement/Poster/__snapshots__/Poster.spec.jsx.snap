// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster component renders placeholder if no passed props 1`] = `
<img
  alt="Poster of No title provided"
  src="http://via.placeholder.com/500x750"
  styleName="poster"
/>
`;

exports[`Poster component renders release year 1`] = `
<img
  alt="Poster of Test title"
  src="http://url.for.test.title.cover"
  styleName="poster"
/>
`;
